marklogic-data-hub-central/README.md

```diff
@@ -6,27 +6,72 @@ Boot middle tier instead of a Node middle tier
 1. For production, you want to deploy an executable Spring Boot jar that includes all of the static UI resources
 
 
-## Using this during development
+# Developing Data Hub Central
 
 During development, you'll run two separate processes - the Spring Boot middle tier (which defaults to Tomcat), and 
-the UI. Node is still used for building and running the UI. 
-
-To run the UI, first go into the ./marklogic-data-hub-central/ui directory:
-
-    cd marklogic-data-hub-central/ui
-
-The first time you run the UI, and any time after you've made changes or pulled down changes, you'll need to run 
-the following task to pull down all of the UI dependencies:
-
-    npm install
+the UI. Node is still used for building and running the UI. In addition, you'll also need to have a Data Hub instance 
+that you can talk to. Running each of these processes is described below. 
+
+## Running a Data Hub instance
+
+If you're developing locally and not connecting to a Data Hub instance elsewhere - such as hosted by DHS - you'll need 
+to have a Data Hub instance running locally that you can connect to. You'll typically want to build this instance off 
+the branch that you're working on so that it has the latest changes from the develop branch. 
+
+The CONTRIBUTING.md for this repository discusses how to do this in the "Testing changes to the Data Hub Gradle plugin" section. 
+This process is also explained in detail below for convenience when working on Data Hub Central. 
+
+First, run this command from the root directory in this repository to publish the Data Hub Gradle plugin to your local 
+Maven repository (defaults to ~/.m2/repository, but you won't need to access that directly):
+
+    ./gradlew publishToMavenLocal -PskipWeb=
+
+Every time you pull in new changes to this repository, you should run that command to ensure you are working with the 
```

Comment:
"Every time you pull in new changes to this repository"... Does this include any changes whatsoever to the marklogic-data-hub repo (Java core, middle tier, UI) or just certain types of changes? For example, I don't think I need to do this when I'm developing the UI.

Reply:
My experience is that this most often occurs when someone is starting a new branch off develop or rebasing their feature branch off a just-updated develop branch. Most developers I've worked with in the field don't want to analyze all the changes they brought in to determine how they should update their app - they just do the "full update" and grab a cup of coffee while that's doing its thing. 

In terms of dh-gradle, a relevant change would be anything in DH core, and/or anything in the Java portion of hub central.